profiler/src/ProfilerEngine/Datadog.Profiler.Native.Linux/OsSpecificApi.cpp

```diff
@@ -178,5 +182,36 @@ int32_t GetProcessorCount()
     return get_nprocs();
 }
 
+std::vector<std::shared_ptr<IThreadInfo>> GetProcessThreads()
+{
+    DIR* proc_dir;
+    char dirname[100] = "/proc/self/task";
+    proc_dir = opendir(dirname);
+
+    std::vector<std::shared_ptr<IThreadInfo>> threads;
+
+    if (proc_dir != nullptr)
+    {
+        on_leave { closedir(proc_dir); };
+        threads.reserve(512);
+
+        /* /proc available, iterate through tasks... */
+        struct dirent* entry;
+        while ((entry = readdir(proc_dir)) != nullptr)
+        {
+            if (entry->d_name[0] == '.')
+                continue;
+            auto threadId = atoi(entry->d_name);
+            threads.push_back(std::make_shared<LinuxThreadInfo>(threadId, OpSysTools::GetNativeThreadName(threadId)));
+        }
+    }
+    else
+    {
+        auto errorNumber = errno;
+        Log::Debug("Failed at opendir ", dirname, " error: ", strerror(errorNumber));
```

Comment:
FAR: emit the message as an error and only once

Reply:
✔️